lib/integration/framework/Test/Integration/Framework/DSL/Wallet.hs

```diff
@@ -0,0 +1,196 @@
+    ( (.~)
+    , (^.)
+    )
+import Data.Text
+    ( Text
+    )
+import Numeric.Natural
+    ( Natural
+    )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , eventually
+    , fixturePassphrase
+    , shouldBe
+    )
+import Test.Integration.Framework.DSL.TestM
+    ( Over
+    , TestM
+    , check
+    , over
+    , partialRequest
+    )
+
+import qualified Cardano.Faucet.Mnemonics as Mnemonics
+import qualified Cardano.Wallet.Api.Clients.Testnet.Shelley as C
+import Cardano.Wallet.Api.Types.Amount
+    ( ApiAmount (..)
+    )
+import Cardano.Wallet.Api.Types.WalletAssets
+    ( ApiWalletAssets (..)
+    )
+import Cardano.Wallet.Faucet
+    ( Faucet (nextShelleyMnemonic)
+    )
+import Cardano.Wallet.Primitive.Types.Tx.TxMeta
+    ( TxStatus (InLedger)
+    )
+
+type AWallet = ApiT WalletId
+
+type Patch a = a -> a
+
+createAWallet
+    :: SomeMnemonic
+    -> (WalletPostData -> WalletPostData)
+    -> TestM AWallet
+createAWallet m15 refine = do
+    apiWallet' <-
+        partialRequest
+            $ C.postWallet
+            $ WalletOrAccountPostData
+            $ Left
+            $ refine
+            $ WalletPostData
+                { addressPoolGap = Nothing
+                , mnemonicSentence = ApiMnemonicT m15
+                , mnemonicSecondFactor = Nothing
+                , name = ApiT $ unsafeFromText "Wallet from mnemonic"
+                , passphrase = ApiT $ unsafeFromText fixturePassphrase
+                , oneChangeAddressMode = Nothing
+                , restorationMode = Nothing
+                }
+    pure $ apiWallet' ^. #id
+
+createARandomWallet :: Patch WalletPostData -> TestM AWallet
+createARandomWallet refine = fst <$> createARandomWalletWithMnemonics refine
+
+createARandomWalletWithMnemonics
+    :: Patch WalletPostData
+    -- ^ Refine the wallet data
+    -> TestM (AWallet, SomeMnemonic)
+createARandomWalletWithMnemonics refine = do
+    m15 <- Mnemonics.generateSome Mnemonics.M15
+    w <- createAWallet m15 refine
+    pure (w, m15)
+
+deleteWallet :: Over AWallet ()
+deleteWallet = do
+    w <- ask
+    void $ lift $ partialRequest $ C.deleteWallet w
+
+waitUntilStateIsReady :: Over AWallet ()
+waitUntilStateIsReady =
+    eventually "Wallet state is ready" $ withApiWallet $ statusIs Ready
+
+balanceIs :: Natural -> Over ApiWallet ()
+balanceIs q = check
+    $ \w -> w ^. #balance . #available . #toNatural `shouldBe` q
+
+statusIs :: SyncProgress -> Over ApiWallet ()
+statusIs expected = check
+    $ \w -> w ^. #state . #getApiT `shouldBe` expected
+
+named :: Text -> Patch WalletPostData
+named name' = #name .~ ApiT (unsafeFromText name')
+
+aFaucetWallet :: TestM AWallet
+aFaucetWallet = do
+    faucet <- asks _faucet
+    faucetMnemonic <- liftIO $ nextShelleyMnemonic faucet
+    faucetWalletId <- createAWallet faucetMnemonic $ named "Faucet wallet"
+    over faucetWalletId waitUntilStateIsReady
+    pure faucetWalletId
+
+fundWallet :: Natural -> Over AWallet ()
```

Comment:
we have a possible overlap with functions in DSL.hs. Also when it comes to `submitTx`

Reply:
Maybe it would be good to add ALSO fund and submit without waiting for successful result (ie. ledger accommodation). Use them in these functions and clean DLS.hs

Reply:
Yes DSL.hs has everything, but I'm trying to move away from there slowly for a simpler approach not involving the  Link module.
On the duplication, I'm not sure we have the man power to touch the old tests. When the new DSL is tested and we like it, we can consider a migration. Not against it. For now , duplication is the safest option.